src/Raven.Server/RavenServer.cs

```diff
@@ -1615,6 +1616,23 @@ public AuthenticationStatus Status
                 set => _status = value;
             }
 
+            public AuthorizationStatus GetAuthorizationStatus(string databaseName)
+            {
+                switch (Status)
+                {
+                    case AuthenticationStatus.ClusterAdmin:
+                        return AuthorizationStatus.ClusterAdmin;
+                    case AuthenticationStatus.Operator:
+                        return AuthorizationStatus.Operator;
+                    case AuthenticationStatus.Allowed:
+                        if (AuthorizedDatabases.TryGetValue(databaseName, out var databaseAccess) && (databaseAccess == DatabaseAccess.Admin))
+                            return AuthorizationStatus.DatabaseAdmin;
+                        return AuthorizationStatus.ValidUser;
+                    default:
+                        return AuthorizationStatus.UnauthenticatedClients;
```

Comment:
Is it safe approach to have `default` here? Since it's sensitive check maybe better to handle explicitly all possible cases?

Reply:
Done